server/src/main/java/org/apache/gravitino/server/web/rest/MetadataObjectCredentialOperations.java

```diff
@@ -0,0 +1,100 @@
+ *  under the License.
+ */
+
+package org.apache.gravitino.server.web.rest;
+
+import com.codahale.metrics.annotation.ResponseMetered;
+import com.codahale.metrics.annotation.Timed;
+import java.util.List;
+import java.util.Locale;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.MetadataObjects;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.catalog.CredentialOperationDispatcher;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.dto.credential.CredentialDTO;
+import org.apache.gravitino.dto.responses.CredentialResponse;
+import org.apache.gravitino.dto.util.DTOConverters;
+import org.apache.gravitino.metrics.MetricNames;
+import org.apache.gravitino.server.web.Utils;
+import org.apache.gravitino.utils.MetadataObjectUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/metalakes/{metalake}/objects/{type}/{fullName}/credentials")
+public class MetadataObjectCredentialOperations {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(MetadataObjectCredentialOperations.class);
+
+  private CredentialOperationDispatcher dispatcher;
+
+  @SuppressWarnings("unused")
+  @Context
+  private HttpServletRequest httpRequest;
+
+  @Inject
+  public MetadataObjectCredentialOperations(CredentialOperationDispatcher dispatcher) {
+    this.dispatcher = dispatcher;
+  }
+
+  @GET
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-credentials." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-credentials", absolute = true)
+  public Response getCredentials(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName) {
+    LOG.info(
+        "Received get credentials request for object type: {}, full name: {} under metalake: {}",
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            MetadataObject object =
+                MetadataObjects.parse(
+                    fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+
+            NameIdentifier identifier = MetadataObjectUtil.toEntityIdent(metalake, object);
+            List<Credential> credentials = dispatcher.getCredentials(identifier);
+            if (credentials == null) {
+              return Utils.ok(new CredentialResponse(new CredentialDTO[0]));
+            }
+            return Utils.ok(
+                new CredentialResponse(
+                    DTOConverters.toDTO(credentials.toArray(new Credential[credentials.size()]))));
+          });
+    } catch (Exception e) {
```

Comment:
Are we sure that this exception will not contain any sensitive information?
The control flow shows that some of these information are logged and printed.

Reply:
yes, the sensitive credential information is not logged.